packages/rum-core/src/domain/tracing/tracer.spec.ts

```diff
@@ -207,6 +207,51 @@ describe('tracer', () => {
       expect(xhrStub.headers['X-B3-TraceId']).toBeUndefined()
     })
 
+    it('should not add any headers when trace not sampled and config set to sampled', () => {
+      const configurationWithInjectionParam = {
+        ...configuration,
+        traceSampleRate: 0,
+        traceContextInjection: TraceContextInjection.SAMPLED,
+      }
+
+      const tracer = startTracer(configurationWithInjectionParam, sessionManager)
+      const context = { ...ALLOWED_DOMAIN_CONTEXT }
+      tracer.traceXhr(context, xhrStub as unknown as XMLHttpRequest)
+
+      expect(xhrStub.headers['x-datadog-trace-id']).toBeUndefined()
+      expect(xhrStub.headers['x-datadog-sampling-priority']).toBeUndefined()
+    })
+
+    it('should add headers when trace sampled and config set to Sampled', () => {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏case consistency with other test descriptions
```suggestion
    it('should add headers when trace sampled and config set to sampled', () => {
```